src/box/wal.c

```diff
@@ -63,6 +63,8 @@ const char *wal_mode_STRS[WAL_MODE_MAX] = {
 
 int wal_dir_lock = -1;
 
+struct rlist wal_on_batch_write = RLIST_HEAD_INITIALIZER(wal_on_batch_write);
```

Comment:
You might want to destroy this list in `wal_free()`.

Reply:
Thanks!
```diff
diff --git a/src/box/wal.c b/src/box/wal.c
index dae8e5f1d..aa9440682 100644
--- a/src/box/wal.c
+++ b/src/box/wal.c
@@ -594,6 +594,8 @@ wal_free(void)
                panic_syserror("WAL writer: thread join failed");
        }
 
+       trigger_destroy(&wal_on_batch_write);
+
        wal_writer_destroy(writer);
 }
 
```